src/polyswarm_api/resources.py

```diff
@@ -1119,3 +1119,99 @@ def __init__(self, content, api=None):
         self.source = content['source']
         self.team_account_id = content['team_account_id']
         self.user_account_id = content['user_account_id']
+
+
+class ReportTask(core.BaseJsonResource):
+    RESOURCE_ENDPOINT = "/reports"
+
+    def __init__(self, content, api=None):
+        super().__init__(content, api=api)
+        self.id = content['id']
+        self.type = content['type']
+        self.format = content['format']
+        self.state = content['state']
+        self.community = content['community']
+        self.created = content['created']
+        self.template_id = content.get('template_id')
+        self.template_metadata = content.get('template_metadata', {})
+        self.sandbox_task_id = content.get('sandbox_task_id')
+        self.instance_id = content.get('instance_id')
+        self.url = content['url']
+
+    def download_report(self, folder=None):
+        """ This method is special, in that it is simply for downloading from S3 """
+        return core.PolyswarmRequest(
+            self.api,
+            {
+                'method': 'GET',
+                'url': self.url,
+                'stream': True,
+                'headers': {'Authorization': None}
+            },
+            result_parser=LocalArtifact,
+            folder=folder,
+        ).execute()
+
+
+class ReportTemplate(core.BaseJsonResource):
+    RESOURCE_ENDPOINT = "/reports/templates"
+
+    def __init__(self, content, api=None):
+        super().__init__(content, api=api)
+        if content:
+            self.id = content['id']
+            self.created = content['created']
+            self.template_name = content['template_name']
+            self.includes = content.get('includes')
+            self.excludes = content.get('excludes')
+            self.primary_color = content.get('primary_color')
+            self.footer_text = content.get('footer_text')
+            self.last_page_text = content.get('last_page_text')
+            self.is_default = content.get('is_default', False)
+            self.logo_content_length = content.get('logo_content_length')
+            self.logo_url = core.concatenate_api_uri(self.api.uri, content.get('logo_url'))
+            self.logo_content_type = content.get('logo_content_type')
+            self.logo_height = content.get('logo_height')
+            self.logo_width = content.get('logo_width')
+
+    def download_logo(self, folder):
+        return core.PolyswarmRequest(
+            self.api,
+            {
+                'method': 'GET',
+                'url': self.logo_url,
+            },
+            result_parser=LocalArtifact,
+            folder=folder,
+        ).execute()
+
+    def delete_logo(self):
+        return core.PolyswarmRequest(
+            self.api,
+            {
+                'method': 'DELETE',
+                'url': self.logo_url,
+            },
+        ).execute()
+
+    def upload_logo(self, logo_file, content_tpe):
```

Comment:
Should it be `content_type`?

Reply:
Yes, the header `Content-Type` has to be set when uploading a file in the API, only `image/jpeg` or `image/png` are allowed, and it's a quick validation of what the user tries to upload (like, when using REST `application/json` is passed instead), although before we save the image we still check it's indeed a PNG/JPEG image.

The automatically sets the header anyway based on the file extension.

Reply:
I mean, I think there is a typo here, it is written `content_tpe`, not `content_type`.